lib/tests/test_core.c

```diff
@@ -130,6 +131,45 @@ test_tskit_version(void)
     CU_ASSERT_EQUAL(TSK_VERSION_PATCH, 15);
 }
 
+static void
+test_gsl_ran_flat_patch(void)
+{
+    double left, right, x;
+    int i;
+    gsl_rng *rng = gsl_rng_alloc(gsl_rng_default);
+
+    // reproduce documented bug using RNG seed+state
+    gsl_rng_set(rng, 18112);
+    FILE *rng_state;
+    rng_state = fopen("../tests/test_core_rngstate0.bin", "r");
+    gsl_rng_fread(rng_state, rng);
+    fclose(rng_state);
+    left = 85189995.000000;
+    right = 85190021.000000;
+    x = msp_gsl_ran_flat(rng, left, right);
+    CU_ASSERT(left <= x);
+    CU_ASSERT(right > x);
```

Comment:
it might be good to run this to _create_ the error too. e.g.,
```suggestion
    x = gsl_ran_flat(rng, left, right);
    CU_ASSERT(left <= x);
    CU_ASSERT(!(right > x));
```